### PR TITLE
fix: Upgrade Go to v1.21 Fix #11556

### LINKED ISSUE #11556: go 1.21 upgrade
upgrade to go 1.21 https://go.dev/doc/go1.21

### PR DESCRIPTION
**Pull request by junkmm**
Fixes #11556 

### Motivation

upgrade to go 1.21

### Modifications

I upgraded the go version by referring to the previous case PR #11027 

### Verification

Checking because the make test failed.

### CODE REVIEW

File: server/static/files.go
Comment (terrytangyuan) [inline server/static/files.go:1]: Why is this file changed?

**Comment by agilgur5** [comment]
Replaced by #11601
